### ISSUE TITLE
yAxis.stackLabels should compute total of positive and negative series

### ISSUE THREAD

**Issue by Madhu94** [highcharts/highcharts#8843]
#### Expected behaviour
``yAxis.stackLabels`` option on column charts should allow ability to sum positive and negative values and show a net value. In the linked  fiddle, there should be a way to make the ``Grapes`` series sum up to -3. 

#### Actual behaviour
The positive and negative stacks are treated separately (``Grapes`` stack adds to 4) , and there is no option to change this behavior.

#### Live demo with steps to reproduce
https://jsfiddle.net/qh4of0gw/3/

#### Product version
Highcharts - 6.1.0

#### Affected browser(s)
All

Please let me know if there already exists an option to get this behavior.

**Comment by pawelfus** [comment]
Hi @Madhu94 

Thanks for reporting the issue. There's no direct option like "merge" positive and negative stacks. Those are always independent, but we can change what is displayed in a stacklabel using formatter: https://jsfiddle.net/BlackLabel/qh4of0gw/10/

Snippet:

```javascript
    yAxis: {
      stackLabels: {
        enabled: true,
        formatter: function() {
          var negativeStack = this.axis.stacks['-column'][this.x];

          return this.isNegative ? '' : this.total + (negativeStack ? negativeStack.total : 0);
        }
      }
    },
```

The way to get negative stack is not obvious, probably we could add a reference to the negative stack (via arguments). What do you think? Something like this:

```
    yAxis: {
      stackLabels: {
        enabled: true,
        formatter: function(oppositeStack) {
          return this.isNegative ? '' : this.total + (oppositeStack ? oppositeStack.total : 0);
        }
      }
    },
```

**Comment by Madhu94** [comment]
@pawelfus  Adding a reference to get the opposite stack sounds great ! It is not obvious that this.axis.stacks['-column'] is how you get it. 

I will use your workaround for now.
Thank you for your help!

**Comment by mlucool** [comment]
@pawelfus Do you have any ideas for how to solve tooltip merging too?

**Comment by pawelfus** [comment]
Hi @mlucool 

You can use the same logic in [`tooltip.formatter`](https://api.highcharts.com/highcharts/tooltip.formatter).

**Comment by Madhu94** [comment]
@pawelfus Will the solution proposed above will be added into Highcharts ?
This - 

   ```
yAxis: {
      stackLabels: {
        enabled: true,
        formatter: function(oppositeStack) {
          return this.isNegative ? '' : this.total + (oppositeStack ? oppositeStack.total : 0);
        }
      }
    },
```

**Comment by TorsteinHonsi** [comment]
Just a side note - if you want to treat positive and negative numbers combined in a stack, it may be better to use a stacked area: https://jsfiddle.net/highcharts/qh4of0gw/14/.

**Comment by mlucool** [comment]
@TorsteinHonsi - is there a plan to add this into highcharts?

**Comment by pawelfus** [comment]
No plans to add this to Highcharts, not at this moment. It depends on a popularity of the enhancement. It may be added to Highcharts in future releases though.

**Comment by govinda18** [comment]
Hi @pawelfus,
I think the above solution no longer works with highcharts@v9 as the axis does not have a stacks property anymore. Can you please suggest an alternative?

**Comment by pawelfus** [comment]
Nice catch! The whole `stacks` logic was moved to a composition `axis.stacking`. Fixed demo: https://jsfiddle.net/BlackLabel/uLrpzgbm/1/

Snippet:

```js
      formatter: function() {
        var negativeStack = this.axis.stacking.stacks['-column'];

        return this.isNegative ? '' : this.total + (negativeStack && negativeStack[this.x] ? negativeStack[this.x].total : 0);
      },
```

FYI @TorsteinHonsi - a good example related to https://github.com/highcharts/highcharts/issues/15933

**Comment by govinda18** [comment]
Thanks a lot @pawelfus! This works.

**Comment by nishikantparmariam** [comment]
@pawelfus this does not seem to work with Highcharts v10.3.3, this path is not valid - 

```js
this.axis.stacking.stacks['-column']
```

instead it is

```js
this.axis.stacking.stacks[`-column,${this.stack || ''},,`];
```

Are there any other ways in which it can be dynamic, I see the other two comma, what are they for?


Reproducer - https://jsfiddle.net/p5m2c809/

**Comment by karolkolodziej** [comment]
Hi @nishikantparmariam! 

Your approach looks valid but if you want to make it more dynamic you can reverse how it was created with: 
https://github.com/highcharts/highcharts/blob/d32d0735096f827982162b57d8f65cb3202f955a/ts/Core/Axis/Stacking/StackingAxis.ts#L131-L155

**Comment by pawelfus** [comment]
Yes, it is. Only one x-axis can be connected to a series. The problem is not xAxis, but `idx` in your example. You can find available series indices from `this.points` object. Example: https://jsfiddle.net/BlackLabel/q8hktp1o/ 

```ts
// Keys in `point` consist of indexes of series and other key-indices, use only series-related ones:
const indices = Object.keys(this.points).filter(key => !key.match(/,/g));
     
// Use 0 or any other index. indices array contains all series indices related to current Stack
const xAxis = this.axis.series[indices[0]].xAxis;
```

**Comment by charlieflipside** [comment]
Just following up on a more generalized example of this. We have a corporate license for highcharts that enables our end users to generate visuals using highcharts; for browser security reasons we do not allow any custom formatters. We identified `{point.total}` method is unable to mix positive and negative values for stacked bar charts. Is there not a more simple way forward (like the 2012 fix to match point.stackTotal and point.total) to create a {point.sumTotal} method that sums across signs (not separating positive and negative values).

Included is an example for how a user's stacked bars only correctly calculate the net total when all values are the same sign (top) but not when signs differ (bottom).


![image](https://github.com/user-attachments/assets/24fb3745-18a4-4048-8d7f-b461b5c45992)
